### ISSUE TITLE
v4/v5 regression in interaction between relationships & `retrieveQuery`

### ISSUE THREAD

**Issue by davidAtInleague** [coldbox-modules/quick#194]
after a `retrieveQuery` in v4, subsequent calls `first` would return a struct, whereas they now seem to return an instance of the underlying entity. Behavior seems unchanged when relationships are not involved.

entities:
```
component table="RMME_A" extends="quick.models.BaseEntity" accessors=true {
    property name="ID_A" type="numeric" sqltype="integer";
    property name="ID_B" type="numeric" sqltype="integer";

    variables._key = "ID_A";

    function B() {
		return belongsTo(
			relationName = "RMME_B",
			foreignKey = "ID_B",
			localKey = "ID_B"
		);
	}
}

component table="RMME_B" extends="quick.models.BaseEntity" accessors=true {
    property name="ID_B" type="numeric" sqltype="integer";

    variables._key = "ID_B";
}
```

repro
```
v = getInstance("RMME_A") // RMME_A
	.retrieveQuery()      // modules.quick.models.QuickQB
	.select("ID_A")       // modules.quick.models.QuickQB
	.first()              // struct (ordered)
	.ID_A;                // number (ok!)

v = getInstance("RMME_A") // RMME_A
	.firstOrFail()        // RMME_A
	.B()                  // quick.models.Relationships.BelongsTo
	.retrieveQuery()      // quick.models.QuickBuilder
	.select("ID_B")       // quick.models.QuickBuilder
	.first()              // RMME_B
	.ID_B;                // component RMME_B has no accessible member "ID_B" (expected a number here)

```

**Comment by ryanalbrecht** [comment]
Bumping this. Can confirm the issue.

**Comment by elpete** [comment]
Fixed in `quick@5.2.1`
